prow/scripts/cluster-integration/kyma-gke-upgrade.sh

```diff
@@ -68,463 +79,399 @@ export EXTERNAL_SOLUTION_TEST_RELEASE_NAME="${EXTERNAL_SOLUTION_TEST_NAMESPACE}"
 export EXTERNAL_SOLUTION_TEST_RESOURCE_LABEL="kyma-project.io/external-solution-e2e-test"
 export TEST_RESOURCE_LABEL_VALUE_PREPARE="prepareData"
 export HELM_TIMEOUT_SEC=10000s # timeout in sec for helm install/test operation
-export TEST_TIMEOUT_SEC=600   # timeout in sec for test pods until they reach the terminating state
+export TEST_TIMEOUT_SEC=600    # timeout in sec for test pods until they reach the terminating state
 export TEST_CONTAINER_NAME="tests"
 
 TMP_DIR=$(mktemp -d)
+KYMA_LABEL_PREFIX="kyma-project.io"
+KYMA_TEST_LABEL_PREFIX="${KYMA_LABEL_PREFIX}/test"
+BEFORE_UPGRADE_LABEL_QUERY="${KYMA_TEST_LABEL_PREFIX}.before-upgrade=true"
```

Comment:
I wonder if we should label the tests like that:
`kyma.project.io/test/upgarde-execution=after || before`
It would be easier to list all upgrade test, example: `--label kyma.project.io/test/upgarde-execution`, but we should consider if we need such functionality.

Reply:
there is really no value in finding all upgrade tests imho. You should not run the verification phase (after upgrade) without the preparation phase (before-upgrade). They need to be executed in that particular order.

But this is also just the first phase of implementing those labels, so it is something that can be easily changed once we figure out that this feature is missing.